keps/sig-api-machinery/4020-unknown-version-interoperability-proxy/README.md

```diff
@@ -240,50 +241,28 @@ This might be a good place to talk about core concepts and how they relate.
 
-1. A new filter will be added to the [handler chain] of the aggregation layer. This filter will maintain an internal map with the key being the group-version-resource and the value being a list of server IDs of apiservers that are capable of serving that group-version-resource
-   1. This internal map is populated using an informer for StorageVersion objects. An event handler will be added for this informer that will get the apiserver ID of the requested group-version-resource and update the internal map accordingly
+1. A new filter will be added to the [handler chain] of the aggregation layer. This filter will maintain the following internal caches:
+
+    1. a map that stores the resources served by the local apiserver for a group-version(Local Discovery cache). This will be done via a discovery call using a loopback client. A post-start hook will populate this cache, guaranteeing the apiserver has a complete view of its served resources before processing any incoming requests.
+    2. an informer cache of resources served by each peer apiserver in the cluster(Remote Discovery cache). This cache will be updated by an informer on [apiserver identity Lease objects](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1965-kube-apiserver-identity/README.md#proposal). The informer's event handler will make remote discovery calls to peer apiservers whose lease objects are added or updated. 
 
 2. This filter will pass on the request to the next handler in the local aggregator chain, if:
    1. It is a non resource request
-   2. The StorageVersion informer cache hasn't synced yet or if `StorageVersionManager.Completed()` has returned false. We will serve error 503 in this case
+   2. The Local Discovery cache or the Remote Discovery cache hasn't synced yet. We will serve error 503 in this case
    3. The request has a header `X-Kubernetes-APIServer-Rerouted:true` that indicates that this request has been proxied once already. If for some reason the resource is not found locally, we will serve error 503
-   4. No StorageVersion was retrieved for it, meaning the request is for an aggregated API or for a custom resource
-   5. If the local apiserver ID is found in the list of serviceable-by server IDs from the internal map
+   4. The requested resource was listed in the Local Discovery cache
+   5. No other peer apiservers were found to exist in the cluster
 
-3. If the local apiserver ID is not found in the list of serviceable-by server IDs, a random apiserver ID will be selected from the retrieved list and the request will be proxied to this apiserver
+3. If the requested resource was not found in the Local Discovery cache, we will try to fetch the resource from the Remote Discovery cache. The request will then be proxied to any peer apiserver thats found to be able to serve the resource as indicated in the Remote Discovery cache. 
```

Comment:
If both the other apiservers can serve, is the selection random? (Assuming so, would be good to explicitly state).

Reply:
Done